internal/exec/stages/fetch_offline/fetch_offline_test.go

```diff
@@ -29,62 +29,107 @@ func checkNeedsNet(t *testing.T, cfg *types.Config) bool {
 	return needsNet
 }
 
-func assertNeedsNet(t *testing.T, cfg *types.Config) {
-	assert.Equal(t, checkNeedsNet(t, cfg), true, "unexpected config doesn't need net: %v", *cfg)
-}
+func TestConfigNotNeedsNet(t *testing.T) {
```

Comment:
Would it make sense to split the breaking into more proper subtests as a preparatory commit and then in the commit that fixes the custom case, we add the new test cases that check it?